janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -212,6 +214,111 @@ private static void assertStorageExists(Backend backend, boolean exists) throws
+        mgmt.buildIndex("mi", Vertex.class).addKey(houses).addKey(deposit).addKey(uuid).buildMixedIndex(INDEX);
+        mgmt.buildIndex("theIndex", Vertex.class).addKey(address).addKey(deposit).addKey(byteKey)
+            .addKey(cars).addKey(birthday).buildMixedIndex(INDEX);
+        finishSchema();
+
+        JanusGraphVertex v = tx.addVertex(T.label, "person", "name", "", "age", 30, "sex", "male", "houses", 0,
+            "cars", 0, "deposit", 0.0, "birthday", new Date(), "instant", Instant.ofEpochMilli(1), "uuid", UUID.randomUUID());
+        v.property("age").property("flag", true);
+        tx.addVertex(T.label, "person", "name", "robert", "nickname", "bob", "sex", "female", "phone", 12345678L,
+            "deposit", 100000.5, "instant", Instant.ofEpochMilli(100), "uuid", UUID.randomUUID(), "single", true);
+        tx.addVertex(T.label, "person", "nickname", "anonym is my name", "phone", 23456789L,
+            "byte", Byte.MIN_VALUE, "uuid", UUID.randomUUID());
+        tx.addVertex(T.label, "person", "houses", 2, "cars", 1, "address", Geoshape.point(37.97, 23.72));
+        tx.commit();
+
+        /* force index to be used */
+        clopen(option(FORCE_INDEX_USAGE), true);
+
+        // test has(key) -> has(key, NOT_EQUAL, null) (exists query) transformation in JanusGraph GraphCentricQuery
+        assertCount(2, tx.query().has("name").vertices());
+        assertCount(2, tx.query().has("nickname").vertices());
+        assertCount(2, tx.query().has("houses").vertices());
+        assertCount(2, tx.query().has("cars").vertices());
+        assertCount(1, tx.query().has("byte").vertices());
+        assertCount(2, tx.query().has("deposit").vertices());
+        assertCount(1, tx.query().has("birthday").vertices());
+        assertCount(2, tx.query().has("instant").vertices());
+        assertCount(3, tx.query().has("uuid").vertices());
+        assertCount(1, tx.query().has("single").vertices());
+
+        // test has(key) -> has(key, neq(null)) transformation in Gremlin query
+        assertEquals(2, graph.traversal().V().has("name").count().next());
+        assertEquals(2, graph.traversal().V().has("nickname").count().next());
+        assertEquals(2, graph.traversal().V().has("houses").count().next());
+        assertEquals(2, graph.traversal().V().has("cars").count().next());
+        assertEquals(1, graph.traversal().V().has("byte").count().next());
+        assertEquals(2, graph.traversal().V().has("deposit").count().next());
+        assertEquals(1, graph.traversal().V().has("birthday").count().next());
+        assertEquals(2, graph.traversal().V().has("instant").count().next());
+        assertEquals(3, graph.traversal().V().has("uuid").count().next());
+        assertEquals(1, graph.traversal().V().has("single").count().next());
+
+        final String backend = readConfig.get(INDEX_BACKEND, INDEX);
+        if (!backend.equals("lucene") && !backend.equals("solr")) {
```

Comment:
Nit: A nicer solution would be to split this test up into two separate methods and have one of them test `Geoshape` and one everything else but `Geoshape`. You can then @Disable the `Geoshape` test in the Lucene and Solr implementations of `JanusGraphIndexTest`. 
This way, the abstract test class stays free of backend related stuff. For examples, have a look at `JanusGraphTest` and it's implementing classes for different storage backends.